modules/media/ajax/FileUpload.php

```diff
@@ -196,20 +196,31 @@ function viewData()
 function getUploadFields()
 {
 
-    $db =& Database::singleton();
+    $db   = \NDB_Factory::singleton()->database();
+    $user = \User::singleton();
 
     $instruments = $db->pselect(
         "SELECT Test_name FROM test_names ORDER BY Test_name",
         []
     );
-    $candidates  = $db->pselect(
-        "SELECT CandID, PSCID FROM candidate ORDER BY PSCID",
-        []
+    // Select only candidates that have had visit at user's sites
+    $qparam    = array();
+    $candquery = "SELECT DISTINCT PSCID FROM candidate c
+                  LEFT JOIN session s USING (CandID)";
+
+    if (!$user->hasPermission('access_all_profiles')) {
+        $candquery    .= " WHERE FIND_IN_SET(s.CenterID, :cid) ORDER BY PSCID";
```

Comment:
```suggestion
        $candquery    .= " WHERE s.CenterID in (:cid) ORDER BY PSCID";
```
`FIND_IN_SET` in to be used in a `SELECT` clause as it return the index where the value was found. In a `WHERE` clause use `WHERE <attribute> in (<list, of, values>)`

Reply:
i believe we're supposed to be using find_in_set in prepared statements as discussed in the old PR https://github.com/aces/Loris/pull/3697#discussion_r196096201

Reply:
Haven't found the discussion about that it #3697. FIND_IN_SET is less efficient than WHERE x IN (values). This also work with prepared statement as long as your values are in a comma separated string which can be acheved with implode()

Reply:
changing my code to what you suggested doesn't give me the expected query. i think the issue is that our Database class pselect function doesn't append the params how you want when using 'IN'

Reply:
Database class should not cause any problem. Will approve this PR and make appropriate change in #4085